### PR TITLE
Add examples

### LINKED ISSUE #27: Add examples to README
Add examples to README as discussed in https://github.com/js-reporters/js-reporters/pull/25#issuecomment-130709569.

### PR DESCRIPTION
**Pull request by flore77**
Fixes #27.

### CODE REVIEW

**Comment by flore77** [comment]
I guess this example should be enough. By adding 2 examples would be too long, maybe adding to this one also a global test. I am not sure.

For an implementation example, I think if we provide links to exiting implementationa in real tools, would be enough, or even better?

File: README.md [hunk 1]
```diff
@@ -79,6 +79,247 @@ For `testStart` and `testEnd`, the corresponding test object is passed to the re
 
 When `runStart`, `suiteStart` and `testStart` are emitted, the `status`, `runtime` and `errors` attributes are `undefined`.
 
+## Example
+
+Lets take the below testsuite and see how the reporting shoud be, based on the standard presented above. For implementations examples please check [Usage](#usage) and [Integrations](#integrations) below for implementations in real tools.
```
Comment (jzaefferer) [inline README.md]: s/shoud/should

s/implementations/implementation (the plural s is already on "examples")

Let's drop "for implementations in real tools", that's redundant.

File: README.md [hunk 2]
```diff
@@ -79,6 +79,247 @@ For `testStart` and `testEnd`, the corresponding test object is passed to the re
...
+  it('test1', function() {
+    expect(true).to.be.true;
+    expect(true).to.be.false;
+  });
+});
+```
+
+The emitted events and data will be in the following order:
+
+- **runStart**:
```
Comment (jzaefferer) [inline README.md]: Having an example at this level of detail makes sense, but its too long for this document. Let's extract that to a separate file (`example.md`? `docs/example.md`?). Maybe we can include a compressed version here, something like...

```md
* **runStart**: `{ name: undefined, childSuites: [Suite('suite1')] }`
* **suiteStart**: `{ name: 'suite1', childSuites: [Suite('suite2')] }`
...
```

File: README.md [hunk 3]
```diff
@@ -79,6 +79,247 @@ For `testStart` and `testEnd`, the corresponding test object is passed to the re
...
+    passed: 1,
+    failed: 1,
+    skipped: 0,
+    total: 2
+  },
+  runtime: 3
+}
+```
+
+## Usage
```
Comment (jzaefferer) [inline README.md]: How about "Usage of the adapters"?

File: README.md [hunk 4]
```diff
@@ -79,6 +79,247 @@ For `testStart` and `testEnd`, the corresponding test object is passed to the re
...
+ } else if (jasmine) {
+  runner = new JsReporters.JasmineAdapter(jasmine.getEnv());
+ }
+ 
+ // Listen to the same events for any testing framework.
+ runner.on('runStart', function(globalSuite) {
+   ...
+ });
+ 
+ runner.on('testEnd', function(test) {
```
Comment (jzaefferer) [inline README.md]: Let's listen to `testEnd` and print failed assertions and `runEnd` and print a summary. That would be a more useful and practical example.

File: README.md [hunk 5]
```diff
@@ -79,6 +79,247 @@ For `testStart` and `testEnd`, the corresponding test object is passed to the re
...
+   ...
+ });
+ 
+ // Or use one of the built-in reporters.
+ JsReporters.TapReporter.init(runner);
+```
+
+## Integrations
+
+ - [BrowserStack](https://github.com/browserstack/browserstack-runner/blob/master/lib/_patch/reporter.js)
```
Comment (jzaefferer) [inline README.md]: Its "browserstack-runner", BrowserStack is the product, browserstack-runner the tool making use of the product.

Comment (flore77) [inline README.md]: Sorry about this, I keep doing this mistake, but browserstack is shorter than browserstack-runner :laughing:

**Comment by flore77** [comment]
@jzaefferer please review

File: README.md [hunk 6]
```diff
@@ -110,6 +110,53 @@ Based on the discussion in [#79](https://github.com/js-reporters/js-reporters/is
 
 Additional properties (not defined here) can be added to the Assertion object.
 
+## Details
+
+For details please check out the [docs](https://github.com/js-reporters/js-reporters/blob/add-examples/docs) and especially the [example](https://github.com/js-reporters/js-reporters/blob/add-examples/docs/example.md) which presents a testsuite and its reporting data based on the standard presented above.
```
Comment (jzaefferer) [inline README.md]: The paths are wrong, need to remove the extra "add-". Might also be better to use relative links, since absolute links "break" for forks.

File: README.md [hunk 7]
```diff
@@ -110,6 +110,53 @@ Based on the discussion in [#79](https://github.com/js-reporters/js-reporters/is
...
+
+For implementations examples please check [Usage of the adapters](#usage-of-the-adapters) and [Integrations](#integrations).
```
Comment (jzaefferer) [inline README.md]: Remove plural S on "implementations"

File: README.md [hunk 8]
```diff
@@ -110,6 +110,53 @@ Based on the discussion in [#79](https://github.com/js-reporters/js-reporters/is
...
+  runner = new JsReporters.JasmineAdapter(jasmine.getEnv());
+}
+ 
+ // Listen to the same events for any testing framework.
+ runner.on('testEnd', function(test) {
+   console.log('Test %s has errors:', test.fullname.join(' '), test.errors);
+ });
+ 
+ runner.on('runEnd', function(globalSuite) {
+   var testCounts = globalSuite.testCounts;
```
Comment (jzaefferer) [inline README.md]: Drop this or use it

File: README.md [hunk 9]
```diff
@@ -110,6 +110,53 @@ Based on the discussion in [#79](https://github.com/js-reporters/js-reporters/is
...
+   
+   console.log('Build status: %d', globalSuite.status);
```
Comment (jzaefferer) [inline README.md]: s/Build/Test? "Build" seems odd.

File: README.md [hunk 10]
```diff
@@ -110,6 +110,53 @@ Based on the discussion in [#79](https://github.com/js-reporters/js-reporters/is
...
+## Usage of the adapters
+
+Listen to the events and receive the emitted data:
+
+```js
+var runner;
+
+// Attach the adapter.
+if (QUnit) {
+  runner = new JsReporters.QUnitAdapter(QUnit);
```
Comment (jzaefferer) [inline README.md]: I wonder if we should add something like `var runner = JsReporters.guess()`, which looks at globals like this if/else/else does and throws an error when it can't find any framework.

**Comment by jzaefferer** [comment]
The readme should linke to `docs/frameworks.md` somewhere.

File: README.md (hunk 8 above)
Comment (flore77) [inline README.md]: I am using it below.

File: README.md (hunk 10 above)
Comment (flore77) [inline README.md]: Yes, sounds good. I will make a separate pr.

File: README.md (hunk 9 above)
Comment (flore77) [inline README.md]: I was thinking to underline how easily is to get the status for the whole testing, like for continuous integration.

Maybe "Testing status" or "Testsuite status", "Test" sounds a little bit confusing, because you we'll think that is the status only of one test.

File: README.md (hunk 8 above)
Comment (jzaefferer) [inline README.md]: Oops, right, nevermind

File: README.md (hunk 9 above)
Comment (jzaefferer) [inline README.md]: "Testsuite status" seems good to me

**Comment by jzaefferer** [comment]
Still missing a direct reference to `docs/frameworks.md` in the README, other looking good.

**Comment by jzaefferer** [comment]
That sounds good!

File: README.md [hunk 11]
```diff
@@ -156,6 +156,17 @@ if (QUnit) {
 
  - [browserstack-runner](https://github.com/browserstack/browserstack-runner/blob/master/lib/_patch/reporter.js) 
 
+## Differences
+
+This section is dedicated to explain the limitations of the adapters in report to the standard.
```
Comment (jzaefferer) [inline README.md]: "in respect"? Instead of "in report"

File: README.md [hunk 12]
```diff
@@ -156,6 +156,17 @@ if (QUnit) {
...
 
+## Differences
+
+This section is dedicated to explain the limitations of the adapters in report to the standard. 
+
+The only limitation is the emitting order, which is not done in source order:
+
+- Jasmine: the emitting order of the tests will be the one from Jasmine
+- Mocha: the emitting order of the tests will be the one from Mocha
+- QUnit: the suites will be emitted in source order, but how a suite was emitted all its tests will be also emitted no matter the source order
```
Comment (jzaefferer) [inline README.md]: I don't understand the part from ",but how", can you rephrase that?

**Comment by flore77** [comment]
@jzaefferer done.

**Comment by jzaefferer** [comment]
Nice, much more clear now. Should be good to land.
